skimage/feature/_sift.pyx

```diff
@@ -0,0 +1,167 @@
+                        np_floats[:, ::1] dist_c,  # const
+                        np_floats rc_bin_spacing):
+    """Compute an array of orientation histograms (Eq. 28 of Otero et. al.)
+
+    Parameters
+    ----------
+    histograms : (n_hist, n_hist, n_ori) ndarray
+        An array of zeros that will contain the histogram output. `n_ori` is
+        the number of orientation bins and `n_hist` is the number of spatial
+        bins along each axis.
+    near_t : (n_patch,) ndarray
+        The orientation histogram bins obtained from `_ori_distances`.
+    near_t_val : (n_patch,) ndarray
+        The orientation histogram values obtained from `_ori_distances`.
+    magnitude : (n_patch,) ndarray
+        The magnitude weights based on spatial distance.
+    dist_r : (n_hist, n_patch) ndarray
+        Row distances between each point in the patch and the nearest row
+        histogram bin. Shape (n_hist, n_patch).
+    dist_c : (n_hist, n_patch) ndarray
+        Column distances between each point in the patch and the nearest column
+        histogram bin.  Shape (n_hist, n_patch).
+    rc_bin_spacing : float
+        The spacing between spatial histogram bins along a single axis.
+
+    """
+    cdef:
+        Py_ssize_t i, p, r, c, k_index, k_index2
+        Py_ssize_t n_patch = len(magnitude)
+        Py_ssize_t n_hist = histograms.shape[0]
+        Py_ssize_t n_ori = histograms.shape[2]
+        np_floats t_val, val_norm1, w0, w1, w2, inv_spacing
+    inv_spacing = 1 / rc_bin_spacing
+    val_norm1 = n_ori / (2 * M_PI)
+    for p in range(n_patch):
+        for r in range(n_hist):
+            if dist_r[r, p] > rc_bin_spacing:
+                continue
+            for c in range(n_hist):
+                if dist_c[c, p] > rc_bin_spacing:
+                    continue
+                w0 = ((1 - inv_spacing * dist_r[r, p])
+                      * (1 - inv_spacing * dist_c[c, p])
+                      * magnitude[p])
+                t_val = near_t_val[p]
+                w1 = w0 * val_norm1 * t_val
+                w2 = w0 - w1
+                k_index = near_t[p]
+                if k_index == n_ori - 1:
+                    k_index2 = 0
+                else:
+                    k_index2 = k_index + 1
+                histograms[r, c, k_index] += w1
+                histograms[r, c, k_index2] += w2
+
+
+cpdef _local_max(np_floats[:, :, ::1] octave, double thresh):
+    cdef:
+        Py_ssize_t n_r = octave.shape[0]
+        Py_ssize_t n_c = octave.shape[1]
+        Py_ssize_t n_s = octave.shape[2]
+        Py_ssize_t r, c, s
+        np_floats* center_ptr
+        Py_ssize_t neighbor_offsets[26]
+        np_floats center_val, val
+        int n = 0
+
+    for r in range(-1, 2):
+        for c in range(-1, 2):
+            for s in range(-1, 2):
+                if (r != 0 or c != 0 or s != 0):
+                    neighbor_offsets[n] = (r * n_c + c) * n_s + s
+                    n += 1
```

Comment:
```suggestion
    for r in range(-1, 2):
        if r == 0:
            continue
        for c in range(-1, 2):
            if c == 0:
                continue
            for s in range(-1, 2):
                if s == 0:
                    continue
                neighbor_offsets[n] = (r * n_c + c) * n_s + s
                n += 1
```

Reply:
I think that change wouldn't work out, because we would skip all 0's, but we actually just want to skip if r, c, and s all are 0.
But if we want to make the if a bit prettier, we could write
`if any([r, c, s]):`

Reply:
:confounded: I confused the `and` and `or` statement, you are right!